cluster/log-dump/log-dump.sh

```diff
@@ -105,20 +103,17 @@ function copy-logs-from-node() {
     # Comma delimit (even the singleton, or scp does the wrong thing), surround by braces.
     local -r scp_files="{$(printf "%s," "${files[@]}")}"
 
-    if [[ -n "${use_custom_instance_list}" ]]; then
+    if [[ "${gcloud_supported_providers}" =~ "${KUBERNETES_PROVIDER}" ]]; then
+      # get-serial-port-output lets you ask for ports 1-4, but currently (11/21/2016) only port 1 contains useful information
+      gcloud compute instances get-serial-port-output --project "${PROJECT}" --zone "${ZONE}" --port 1 "${node}" > "${dir}/serial-1.log" || true
```

Comment:
I wish this wasn't also part of the PR. One issue at at a time, please. :)

Reply:
This is part of the same issue. The point was to use 'gcloud ssh' for all gcloud supported providers (even though there's a custom_instance_list).
And this is the right behaviour - we shouldn't use gcloud just because we're using custom instance list (as the provider might be some non-gcloud one like aws).